x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorFiltering.java

```diff
@@ -194,23 +197,41 @@ public ConnectorFiltering build() {
         }
     }
 
-    public static boolean isDefaultRulePresentInFilteringRules(List<FilteringRule> rules) {
-        FilteringRule defaultRule = getDefaultFilteringRule(null);
-        return rules.stream().anyMatch(rule -> rule.equalsExceptForTimestampsAndOrder(defaultRule));
-    }
-
-    public static FilteringRule getDefaultFilteringRule(Instant timestamp) {
+    public static FilteringRule getDefaultFilteringRule(Instant timestamp, Integer order) {
         return new FilteringRule.Builder().setCreatedAt(timestamp)
             .setField("_")
             .setId("DEFAULT")
-            .setOrder(0)
+            .setOrder(order)
             .setPolicy(FilteringPolicy.INCLUDE)
             .setRule(FilteringRuleCondition.REGEX)
             .setUpdatedAt(timestamp)
             .setValue(".*")
             .build();
     }
 
+    public static List<FilteringRule> sortFilteringRulesByOrder(List<FilteringRule> rules) {
+        if (rules.isEmpty()) {
+            return List.of(getDefaultFilteringRule(null, 0));
+        }
+
+        Optional<FilteringRule> defaultRuleTimeStamp = rules.stream()
+            .filter(rule -> rule.equalsExceptForTimestampsAndOrder(getDefaultFilteringRule(null, 0)))
+            .findFirst();
+
+        List<FilteringRule> sortedRules = rules.stream()
+            .filter(rule -> rule.equalsExceptForTimestampsAndOrder(getDefaultFilteringRule(null, 0)) == false)
```

Comment:
maybe let's create another method `isDefaultFilteringRule`, it might be confusing why we do this:
```
rule.equalsExceptForTimestampsAndOrder(getDefaultFilteringRule(null, 0))
```